lib/utils/architecture-utils.ts

```diff
@@ -0,0 +1,51 @@
+import "reflect-metadata";
+
+export enum ArchType {
+  ARM = "arm",
+  X86 = "x86",
+}
+
+export const addonArchitectureMap = new Map<string,ArchType[]>();
+
+/**
+ * Returns true of false depending on if the passed addon is 
+ * supported by the passed architecture
+ * @param addOn, arch
+ * @returns boolean
+ */
+
+export function isSupportedArchitecture(addOnName: string, arch: ArchType) : boolean | undefined {
+
+  const archs = addonArchitectureMap.get(addOnName);
+  if (archs === undefined) {
+    if (arch === ArchType.X86) {
+      return true;
+    }
+    else {
+      return false;
+    }
+  }
+  return archs.includes(arch);
+}
+
+export function validateSupportedArchitecture(addOnName: string, arch: ArchType) : void {
```

Comment:
let's think if there is a case when let's say there is a partner addon that does not have any `@arch` annotations. And they do support arm for example. I am think we should provide a workaround to force such cases if needed and document. For example, the builder will have a setting like `strictValidation` which is true by default. If not strict, we will issue warning. This boolean flag can be passed to this function as a parameter from the builder.